### PR TITLE
Allow to use Steve's SQLite backed cache

### PR DESCRIPTION
**Pull request by moio**
## Issue: https://github.com/rancher/rancher/issues/45635
Refers to the [Informer-based, SQLite-backed Steve caching ("Vai")](https://confluence.suse.com/pages/viewpage.action?pageId=1359086083) RFC
 
## Problem
Enables Steve's SQLite cache on a feature flag.
 
## Solution
Adds a feature flag to enable Steve's SQLite cache.
 
### Manual Testing
Done.

### Automated Testing
To be filled in with QA references (work in progress)

### CODE REVIEW

**Comment by moio** [comment]
> Recommendation on reverting the test changes and a question on how this works for downstream clusters.

Done, moved to https://github.com/rancher/rancher/pull/45679

File: pkg/features/feature.go [hunk 1]
```diff
@@ -118,6 +118,13 @@ var (
 		"Enable UI Extensions when starting Rancher",
 		true,
 		false,
+		true,
+	)
+	OnDiskSteveCache = newFeature(
```
[the remark this answers is not in the document]
Comment (moio) [reply pkg/features/feature.go]: -> [epic](https://github.com/rancher/rancher/issues/45635)

File: scripts/provisioning-tests [hunk 2]
```diff
@@ -122,6 +122,9 @@ dump_rancher_logs()
   echo -e "\n-----RANCHER-LOG-DUMP-END-----"
 }
 
+# Build Rancher
+./scripts/build-server
```
Comment (snasovich) [inline scripts/provisioning-tests:126]: Good catch but I think it makes sense to explain more on why we're building it as someone without context may remove it with thoughts like "`./scripts/run` includes build so why are we building it separately?"

Maybe also move it to `run_rancher` function and rename it to something like `build_and_run_rancher` - but that's largely opinion-based so fine to ignore it.

File: scripts/provisioning-tests
Comment (snasovich) [inline scripts/provisioning-tests:1]: The same change should be applied to `scripts/test` file.

File: scripts/provisioning-tests (hunk 2 above)
Comment (moio) [reply scripts/provisioning-tests]: addressed as of 29162c8f5

File: scripts/provisioning-tests
Comment (moio) [reply scripts/provisioning-tests:1]: addressed as of 29162c8f5

File: pkg/features/feature.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MbolotSuse) [reply pkg/features/feature.go]: Resolving on the basis that we can fix this after merge, but before 2.9.0 goes live.

**Review by MbolotSuse — APPROVED** [review]
Approving around the changes to the feature objects and how it is used in the context of steve, under the understanding that we will adjust these before 2.9.0 in the near future.

**Review by snasovich — APPROVED** [review]

**Review by tomleb — APPROVED** [review]

**Pull request merged**